### PR TITLE
Massive plumbing layer/placement improvements

### LINKED ISSUE #49660: Fluid Ducts do not connect to other layers using the wrench.
## Reproduction:
1. Place a flud duct on a tile
2. Place another fluid duct on a tile adjacent at a different duct layer.
3. With empty hands, click and drag between both ducts.  You will get a message saying you need a wrench.
4. Put a wrench in your hand and do it again.
5. Nothing happens.

Checked to see if they were connected but just not visually updated and nope, Having one side a Synthesizer and the other a tank, nothing was flowing though any of the pipes.

Be nice if this worked or if you could change the input/output layer on the plumbing devices.  Make it much easier to make smaller networks.

### PR DESCRIPTION
**Pull request by Tastyfish**
## About The Pull Request

Loads of fixed to make layers (all 5!) work. I have likely forgotten a large portion of the minor bugs I kept finding that are since fixed.

### Features:
- All 5 layers.
image

- The machine plumbing connectors now visually connect to the device regardless of layer. (usually looks decent)
image

- As part of doing that, I ended up completely redoing the duct sprite as it always bugged me that the flange blends into a lot of turf plating/underplating sprites.
image

- You can place ducts under machines IF they aren't on the same layer as any of the plumbing connectors.
image

- Ducts can now be placed with the plumbing RCD! Alt-click to choose color and layer.
- Layer can also be selected with the scroll wheel, just like an RPD.
- Ducts can be consumed by RCD's now.

- Machine plumbing connectors now use RGB colors, so you could in theory go nuts. Thinking about it, you could technically have 20 plumbing connectors on 1 machine, each a different color, 1 for each dir and layer combo. (please don't do this for player sanity)

### Bug fixes (Not comprehensive):
Fixes #49660
- Plumbing ducts now have consistent layering, as seen in the "all 5 layers" image, placed between gas pipes and cables.
- (Refactor) Some commonly used, static lists are now in GLOB
- Several places that didn't properly treat the layer bitflags as bitflags, now do.
- Closing out of the plumbing RCD's radial menu no longer sets the blueprint to null.
- Placing a duct where one already was no longer runtimes.
- Unwrenching a duct now preserves the color and layer into the new stack.
- Sanity checks for placing ducts and machines now allow for appropriate layering properly (and in some places, actually check now).
- User is given clearer feedback for multiple placement, etc failure conditions.
- Omni ducts now work as advertised.
- Map-placed colors and layers should now be respected.

Thinking about it, it's crazy that we call air ducts "pipes" and water pipes "ducts". Oh well.

## Why It's Good For The Game

Less cryptic failures and hidden runtimes. 5 layers.

This should be simple to do and now is (ignore that pH buffers aren't necessary here, just a vague demo):
image

## Changelog

:cl:
qol: Plumbing now supports 5 layers. 
qol: The plumbing constructor can now place ducts and change layer via scroll wheel.
fix: Bunch of cryptic failures and errors fixed in placing plumbing.
imageadd: New plumbing duct sprites.
/:cl:

### CODE REVIEW

**Comment by ArcaneMusic** [comment]
Ayo, would you want to consider changing the default layer for hydroponics trays to help in fixing #66359?

**Comment by Tastyfish** [comment]
> Ayo, would you want to consider changing the default layer for hydroponics trays to help in fixing #66359?

As a compromise, I'll fix hydroponics trays as a seperate PR cause there's multiple wonky plumbing things happening with that machine.

**Comment by Ghilker** [comment]
Chemistry - the atmos update

**Comment by Time-Green** [comment]
Thank you

File: code/datums/components/plumbing/_plumbing.dm
Comment (tralezab) [inline code/datums/components/plumbing/_plumbing.dm:173]: better var name than a single letter var, please

File: code/datums/components/plumbing/_plumbing.dm
Comment (tralezab) [inline code/datums/components/plumbing/_plumbing.dm:223]: better var name than a single letter var please

File: code/datums/components/plumbing/_plumbing.dm
Comment (tralezab) [inline code/datums/components/plumbing/_plumbing.dm:219]: and a better var name for this too, because you're using it you may as well improve it while you're here

File: code/game/objects/items/RCD.dm
Comment (tralezab) [inline code/game/objects/items/RCD.dm:1085]: I think it'd be better to put this on the examine proc under a span_notice(), but i'll leave it up to you.

File: code/game/objects/items/RCD.dm
Comment (tralezab) [inline code/game/objects/items/RCD.dm:1182]: if you're going to use the var, may as well improve the var name of `A`

File: code/game/objects/items/RCD.dm
Comment (tralezab) [inline code/game/objects/items/RCD.dm:1204]: T ends up being the most okay single letter var name but still, better var name since you use it and all that (can even just be turf i'd be ok with that but there are probably better ideas)

File: code/game/objects/items/RCD.dm
Comment (tralezab) [inline code/game/objects/items/RCD.dm:1326]: check this first for the early return before delta_y < and > checks. Aka, `if(delta_y == 0) return` then rest of the code

File: code/modules/plumbing/ducts.dm
Comment (tralezab) [inline code/modules/plumbing/ducts.dm:193]: improve the var name since you use it

File: code/modules/plumbing/ducts.dm
Comment (tralezab) [inline code/modules/plumbing/ducts.dm:284]: improve the var name since you use it

File: code/modules/plumbing/ducts.dm
Comment (tralezab) [inline code/modules/plumbing/ducts.dm:287]: and improve this (again, can be as simple as T > turf). machine is fine

File: code/modules/plumbing/ducts.dm
Comment (tralezab) [inline code/modules/plumbing/ducts.dm:314]: better var name, please

File: code/modules/plumbing/ducts.dm
Comment (tralezab) [inline code/modules/plumbing/ducts.dm:369]: swap these checks, check if D is anchored and return if it is, then do the rest of the code unindented.

aka early return

**Review by tralezab — CHANGES_REQUESTED** [review]
god's work with this pr

**Comment by Tastyfish** [comment]
@tralezab In lieu of responding to all ~20 review items, I've done them all, plus compressing some old for loops into `for(...as anything in...)` loops, and a bunch of other 1-letter vars in procs I didn't touch in these types.

File: code/datums/components/plumbing/_plumbing.dm
Comment (SuperNovaa41) [inline code/datums/components/plumbing/_plumbing.dm:199]: You should do an early continue here

File: code/datums/components/plumbing/_plumbing.dm
Comment (SuperNovaa41) [inline code/datums/components/plumbing/_plumbing.dm:228]: here as well

**Review by SuperNovaa41 — CHANGES_REQUESTED** [review]

File: code/datums/components/plumbing/_plumbing.dm
Comment (Tastyfish) [reply code/datums/components/plumbing/_plumbing.dm]: done

File: code/datums/components/plumbing/_plumbing.dm
Comment (Tastyfish) [reply code/datums/components/plumbing/_plumbing.dm]: done

**Comment by Tastyfish** [comment]
waiting for a reviewer

**Comment by LemonInTheDark** [comment]
Runtime's for issues, no point labeling runtime fixes with it

**Comment by Tastyfish** [comment]
still waiting for approval

**Comment by LemonInTheDark** [comment]
@tralezab ?

File: code/datums/components/plumbing/_plumbing.dm
Comment (ShizCalev) [inline code/datums/components/plumbing/_plumbing.dm:26]: while it would be obvious what it is to hex color code readers, let's put that it's red in a comment beside this, and blue for the other

Comment (Tastyfish) [reply code/datums/components/plumbing/_plumbing.dm]: Added for both colors

File: code/datums/components/plumbing/reaction_chamber.dm
Comment (ShizCalev) [inline code/datums/components/plumbing/reaction_chamber.dm:53]: same with these please

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: code/datums/components/plumbing/reaction_chamber.dm
Comment (Tastyfish) [reply code/datums/components/plumbing/reaction_chamber.dm]: Done for reaction chamber too

File: code/game/objects/items/RCD.dm
Comment (ShizCalev) [inline code/game/objects/items/RCD.dm:1216]: is there a reason we went from checking to see if there's anything dense in the tile to just if there's a dense machine in the tile?

Comment (Tastyfish) [reply code/game/objects/items/RCD.dm:1216]: I'm sure I had a reason in my head, but I don't remember what it was now (maybe I was thinking of making the check more complex?). I'll change it back to /obj real quick since I can't justify it.

File: code/game/objects/items/RCD.dm
Comment (ShizCalev) [inline code/game/objects/items/RCD.dm:1255]: src.loc

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: code/game/objects/items/RCD.dm
Comment (ShizCalev) [inline code/game/objects/items/RCD.dm:1277]: src.loc

File: code/game/objects/items/RCD.dm
Comment (ShizCalev) [inline code/game/objects/items/RCD.dm:1296]: .

**Review by ShizCalev — CHANGES_REQUESTED** [review]

**Comment by ShizCalev** [comment]
will look over ducts.dm in the morning, the rest looks good so far to me though.

File: code/game/objects/items/RCD.dm
Comment (Tastyfish) [reply code/game/objects/items/RCD.dm:1216]: Changed to /obj; none of the code relied on it being machines.

File: code/game/objects/items/RCD.dm
Comment (Tastyfish) [reply code/game/objects/items/RCD.dm]: Replaced *all* instanced of `src.loc` in rcd.dm to loc since these playsound lines are just copy-pastes in the first place.

File: code/game/objects/items/RCD.dm
Comment (Tastyfish) [reply code/game/objects/items/RCD.dm]: Replaced *all* instanced of `src.loc` in rcd.dm to loc since these playsound lines are just copy-pastes in the first place.

File: code/game/objects/items/RCD.dm
Comment (Tastyfish) [reply code/game/objects/items/RCD.dm]: Replaced *all* instanced of `src.loc` in rcd.dm to loc since these playsound lines are just copy-pastes in the first place.

**Comment by Tastyfish** [comment]
> will look over ducts.dm in the morning, the rest looks good so far to me though.

Sounds good, thank you kindly. I've made all the changes you've proposed so far.

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: code/modules/plumbing/ducts.dm
Comment (ShizCalev) [inline code/modules/plumbing/ducts.dm:42]: Curious, is there a reason for this var? Do you foresee anything needing to make it false at some point?

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: code/modules/plumbing/ducts.dm
Comment (Tastyfish) [reply code/modules/plumbing/ducts.dm:162]: In the current iteration, no. I'll remove the var.

File: code/modules/plumbing/ducts.dm
Comment (Tastyfish) [reply code/modules/plumbing/ducts.dm:366]: I just realized that, oddly, stack/add() doesn't check against max_amount. However, I've changed that code to the more standard-following form of replacing it with a stack and merging them.

**Pull request merged**
